src/App/Pages/Send/SendAddEditPage.xaml.cs

```diff
@@ -0,0 +1,229 @@
+                    {
+                        var data = message.Data as Tuple<byte[], string>;
+                        _vm.FileData = data.Item1;
+                        _vm.FileName = data.Item2;
+                    });
+                }
+            });
+            await LoadOnAppearedAsync(_scrollView, true, async () =>
+            {
+                var success = await _vm.LoadAsync();
+                if (!success)
+                {
+                    await Navigation.PopModalAsync();
+                    return;
+                }
+                if (!_vm.EditMode && string.IsNullOrWhiteSpace(_vm.Send?.Name))
+                {
+                    RequestFocus(_nameEntry);
+                }
+            });
+        }
+        
+        protected override void OnDisappearing()
+        {
+            base.OnDisappearing();
+            if (Device.RuntimePlatform != Device.iOS)
+            {
+                _broadcasterService.Unsubscribe(nameof(SendAddEditPage));
+            }
+        }
+
+        private void OnMaxAccessCountTextChanged(object sender, TextChangedEventArgs e)
+        {
+            if (string.IsNullOrEmpty(e.NewTextValue))
+            {
+                _vm.MaxAccessCount = null;
+                _maxAccessCountStepper.Value = 0;
+                return;
+            }
+            // accept only digits
+            if (!int.TryParse(e.NewTextValue, out int value))
+            {
+                ((Entry)sender).Text = e.OldTextValue;
+            }
+        }
+        
+        private async void ChooseFile_Clicked(object sender, EventArgs e)
+        {
+            if (DoOnce())
+            {
+                await _vm.ChooseFileAsync();
+            }
+        }
+
+        private void ClearExpirationDate_Clicked(object sender, EventArgs e)
+        {
+            if (DoOnce())
+            {
+                _vm.ClearExpirationDate();
+            }
+        }
+        
+        private async void Save_Clicked(object sender, EventArgs e)
+        {
+            if (DoOnce())
+            {
+                await _vm.SubmitAsync();
+            }
+        }
+        
+        private async void RemovePassword_Clicked(object sender, EventArgs e)
+        {
+            if (DoOnce())
+            {
+                await _vm.RemovePasswordAsync();
+            }
+        }
+        
+        private async void CopyLink_Clicked(object sender, EventArgs e)
+        {
+            if (DoOnce())
+            {
+                await _vm.CopyLinkAsync();
+            }
+        }
+        
+        private async void ShareLink_Clicked(object sender, EventArgs e)
+        {
+            if (DoOnce())
+            {
+                await _vm.ShareLinkAsync();
+            }
+        }
+        
+        private async void Delete_Clicked(object sender, EventArgs e)
+        {
+            if (DoOnce())
+            {
+                if (await _vm.DeleteAsync())
```

Comment:
Are we confirming this with the user first? Not sure how the other clients handle this.

Reply:
Yessir, destructive actions (delete and remove password) prompt for confirmation.  It all happens within DeleteAsync.